python/ray/data/_internal/output_buffer.py

```diff
@@ -31,8 +33,11 @@ class BlockOutputBuffer:
         ...     yield output.next() # doctest: +SKIP
     """
 
-    def __init__(self, target_max_block_size: int):
+    def __init__(
+        self, target_max_block_size: int, target_max_rows_per_block: int = None
```

Comment:
Should check that only one of `target_max_block_size` and `target_max_rows_per_block` is non-None. 
Maybe introduce such a struct to make code cleaner.
```python
@dataclass
class OutputBlockSizeOption:

    target_max_block_size: Optional[int]
    target_max_rows_per_block: Optional[int]

    def __post_init__(sefl):
        # check Nones
```

Reply:
Please fix the type as well